examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/UserLogProcessing.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Spark.Sql;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Spark.Examples.Sql.Batch
+{
+    /// <summary>
+    /// An example demonstrating log processing using user defined functions, regular expressions, and Spark SQL.
+    /// </summary>
+    internal sealed class ApacheUserLogProcessing : IExample
+    {
+        public void Run(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.Error.WriteLine(
+                    "Usage: ApacheUserLogProcessing <path to Apache User Logs>");
+                Environment.Exit(1);
+            }
+
+            SparkSession spark = SparkSession
+                .Builder()
+                .AppName("Apache User Log Processing")
+                .GetOrCreate();
+
+            // Read input log file and display it
+            var df = spark.Read().Text(args[0]);
+            df.Show();
+
+            // Register UDF to perform regex on each row of input file
+            spark.Udf().Register<string, string, bool>("GeneralReg", (log, type) => RegTest(log, type));
```

Comment:
What value does this example have over the Basic we have? @rapoth any thought on this?

Reply:
Through the survey on the .NET blog and customer calls, many customers have expressed interest in log processing and wanted to see an example with it. I tried to make this example more log-specific rather than the more general Basic example. As I’m devising Spark 101 content for the upcoming MS Learn tutorials (to be recorded in September), I wanted an example suitable for beginners but more specific to what users asked for. I also wanted a readme to help walk users through the basic. I can try to rework it if there is advanced Spark.NET functionality you think would make it more useful for those interested in logging?

Reply:
I added some more complexity to this example that uses regex grouping and parses the log file in a more in-depth way so that users can gain even more specific insight on certain columns of their log file. Let me know what you think!

Reply:
As an update, will likely be moving this example to an end-to-end scenario folder (like the PR for the GitHub scenario).